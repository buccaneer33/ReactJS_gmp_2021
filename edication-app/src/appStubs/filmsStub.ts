import {newMovieCard} from '../common/interfaces/ApiDataInterface';

export const filmStub: newMovieCard[] = [
    {
        id: 1000,
        title: 'testTitle',
        tagline: 'testTagline',
        vote_average: 100,
        vote_count: 5,
        release_date: 'testDate',
        poster_path: 'testPoster',
        overview: 'testOverview',
        budget: 100500,
        revenue: 1,
        runtime: 100,
        genres: ['ganre']
    }
];
