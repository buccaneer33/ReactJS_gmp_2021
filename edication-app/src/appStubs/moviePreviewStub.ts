import { MovieCard } from '../common/interfaces/ApiDataInterface';
export const moviePreviewStub: MovieCard[] = [
    {
        id: 1,
        title: 'Pulp Fiction',
        ganre: 'Action & Advanture',
        releaseDate: '2004',
        movieUrl: '',
        overview: '',
        movieRating: 4.5,
        movieDuration: 120,
        movieCardUrl: ''
    },
    {
        id: 2,
        title: 'Bohemian Rhapsody',
        ganre: 'Drama, Biografy, Music',
        releaseDate: '2010',
        movieUrl: '',
        overview: '',
        movieRating: 4.4,
        movieDuration: 136,
        movieCardUrl: ''
    },
    {
        id: 3,
        title: 'Kill Bill',
        ganre: 'Oscar winning Movie',
        releaseDate: '1998',
        movieUrl: '',
        overview: '',
        movieRating: 4.7,
        movieDuration: 173,
        movieCardUrl: ''
    },
    {
        id: 4,
        title: 'Avengers',
        ganre: 'Action & Advanture',
        releaseDate: '2010',
        movieUrl: '',
        overview: '',
        movieRating: 4.5,
        movieDuration: 136,
        movieCardUrl: ''
    },
    {
        id: 5,
        title: 'Inception',
        ganre: 'Action & Advanture',
        releaseDate: '2013',
        movieUrl: '',
        overview: '',
        movieRating: 4.0,
        movieDuration: 181,
        movieCardUrl: ''
    },
    {
        id: 6,
        title: 'Reservoir dogs',
        ganre: 'Oscar winning Movie',
        releaseDate: '1994',
        movieUrl: '',
        overview: '',
        movieRating: 4.8,
        movieDuration: 140,
        movieCardUrl: ''
    }
];
