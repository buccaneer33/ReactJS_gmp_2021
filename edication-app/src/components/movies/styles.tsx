import {createUseStyles} from 'react-jss'

export const stylesList = createUseStyles({
    movies: {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'flex-start',
        justifyContent: 'flex-start',
        width: '100%',
        maxWidth: '1200px',
        backgroundColor: '#232323',
        color: '#F65261',
        fontSize: '19px',
        flexGrow: '1',
        boxSizing: 'border-box',
        padding: '0 50px'
    },
    moviesBlock: {
        display: 'flex',
        flexDirection: 'row',
        alignItems: 'flex-start',
        justifyContent: 'flex-start',
        width: '100%',
        maxWidth: '1200px',
        flexGrow: '1',
        boxSizing: 'border-box',
        flexWrap: 'wrap',
        paddingTop: '25px'
    }
});
