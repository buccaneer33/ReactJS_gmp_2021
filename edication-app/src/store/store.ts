import {createStore, applyMiddleware, compose} from 'redux';
import {rootReducer}  from './redicers/reducers';
import thunk from 'redux-thunk';


const composeEnhancers = window['__REDUX_DEVTOOLS_EXTENSION_COMPOSE__'] as typeof compose || compose;
const persistedState = localStorage.getItem('reduxState') ? JSON.parse(localStorage.getItem('reduxState')) : {}

export const store = createStore(
    rootReducer,
    persistedState,
    composeEnhancers(applyMiddleware(thunk))
);

export type RootState = ReturnType<typeof rootReducer>;
export type AppDispatch = typeof store.dispatch

store.subscribe(() => {
    localStorage.setItem('reduxState', JSON.stringify(store.getState()));
});
