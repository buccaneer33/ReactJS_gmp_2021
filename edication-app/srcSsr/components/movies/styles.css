.movies {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: flex-start;
    width: 100%;
    max-width: 1200px;
    background-color: #232323;
    color: #F65261;
    font-size: 19px;
    flex-grow: 1;
    box-sizing: border-box;
    padding: 0 50px
}
.moviesBlock {
    display: flex;
    flex-direction: row;
    align-items: flex-start;
    justify-content: flex-start;
    width: 100%;
    max-width: 1200px;
    flex-grow: 1;
    box-sizing: border-box;
    flex-wrap: wrap;
    padding-top: 25px
}
.noFoundBlock {
    height: 100%;
    width: 100%
}
.noFoundHeader {
    color: white
}
