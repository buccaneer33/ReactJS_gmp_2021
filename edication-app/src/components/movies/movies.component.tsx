import React from 'react';
import {useStyles} from './styles';
import {NavigationComponent} from '../navigation/navigation.component';
import {MoviesCounterComponent} from '../moviesCounter/moviesCounter.component';
import {MovieItemComponent} from '../movieItem/movieItem.component';
import {ErrorBoundary} from '../../components/errorBoundary/errorBoundary.component';
import {newMovieCard} from '../../common/interfaces/ApiDataInterface';

interface MoviesProps {
    movies: newMovieCard[];
}

export const MoviesComponent: React.FC<MoviesProps> = ({movies}): JSX.Element => {

    const classes = useStyles();

    return (
        <div className={classes.movies}>
            <NavigationComponent />
            <ErrorBoundary>
                {(movies instanceof Array && movies.length) && <MoviesCounterComponent moviesCount={movies.length}/>}
                {(movies instanceof Array && movies.length) && <div className={classes.moviesBlock}>
                    {movies.map((item) => <MovieItemComponent key={item.id} movieCard={item}/>)}
                </div>}
            </ErrorBoundary>
        </div>
    );
}
