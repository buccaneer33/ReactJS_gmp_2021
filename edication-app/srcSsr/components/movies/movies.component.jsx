import React from 'react';
import styles from './styles.css';
import {NavigationComponent} from '../navigation/navigation.component';
import {MoviesCounterComponent} from '../moviesCounter/moviesCounter.component';
import {MovieItemComponent} from '../movieItem/movieItem.component';
import {ErrorBoundary} from '../../components/errorBoundary/errorBoundary.component';

export const MoviesComponent = ({movies}) => {

    return (
        <div class={styles.movies}>
            <NavigationComponent />
            <ErrorBoundary>
                {(movies instanceof Array && movies.length) && <MoviesCounterComponent moviesCount={movies.length}/>}
                {(movies instanceof Array && movies.length) && <div class={styles.moviesBlock}>
                    {movies.map((item) => <MovieItemComponent key={item.id} movieCard={item}/>)}
                </div>}
                {(typeof movies === 'undefined' || (movies instanceof Array && !movies.length)) &&
                <div class={styles.noFoundBlock}>
                    <h2 class={styles.noFoundHeader}>No movies found</h2>
                </div>}
            </ErrorBoundary>
        </div>
    );
}
