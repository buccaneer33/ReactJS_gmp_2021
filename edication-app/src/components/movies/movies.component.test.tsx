import React from 'react';
import {render, screen} from '@testing-library/react';
import {MoviesComponent} from './movies.component';
import {filmStub} from '../../appStubs/filmsStub';
import {store} from '../../store/store';
import {Provider} from "react-redux";
import {
    BrowserRouter as Router,
} from "react-router-dom";

describe('LogoComponent', () => {

    // let component;

    const customRender = () => {
        return render(
            <Provider store={store}>
                <Router>
                    <MoviesComponent movies={filmStub} />
                </Router>
            </Provider>
        ) 
    }
    const emptyRender = () => {
        return render(
            <Provider store={store}>
                <Router>
                    <MoviesComponent movies={[]} />
                </Router>
            </Provider>
        ) 
    }

    test('component conteins "testTitle"', () => {
        customRender();
        expect(screen.getByText('testTitle')).toBeInTheDocument();
    });
    test('component conteins "testDate"', () => {
        customRender();
        expect(screen.getByText('testDate')).toBeInTheDocument();
    });
    test('component with empty rendom', () => {
        emptyRender();
        expect(screen.getByText('No movies found')).toBeInTheDocument();
    });
    test('test snapchot component', () => {
        const component = customRender().container;
        expect(component).toMatchSnapshot();
    });
});
