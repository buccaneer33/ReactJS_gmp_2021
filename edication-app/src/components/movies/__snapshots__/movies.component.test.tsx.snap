// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoComponent test snapchot component 1`] = `
<div>
  <div
    class="movies-0-2-2"
  >
    <div
      class="nav-0-2-6"
    >
      <ul
        class="menu-0-2-7"
      >
        <li
          class="item-0-2-8"
        >
          <button
            class="link-0-2-9"
          >
            all
          </button>
        </li>
        <li
          class="item-0-2-8"
        >
          <button
            class="link-0-2-9"
          >
            Comedy
          </button>
        </li>
        <li
          class="item-0-2-8"
        >
          <button
            class="link-0-2-9"
          >
            Drama
          </button>
        </li>
        <li
          class="item-0-2-8"
        >
          <button
            class="link-0-2-9"
          >
            Romance
          </button>
        </li>
        <li
          class="item-0-2-8"
        >
          <button
            class="link-0-2-9"
          >
            Action
          </button>
        </li>
        <li
          class="item-0-2-8"
        >
          <button
            class="link-0-2-9"
          >
            Adventure
          </button>
        </li>
        <li
          class="item-0-2-8"
        >
          <button
            class="link-0-2-9"
          >
            Fantasy
          </button>
        </li>
      </ul>
      <div
        class="sortBlock-0-2-10"
      >
        <span
          class="title-0-2-11"
        >
          sort by
        </span>
        <select>
          <option>
            release_date
          </option>
          <option>
            genres
          </option>
          <option>
            title
          </option>
        </select>
      </div>
    </div>
    <div
      class="block-0-2-12"
    >
      1
      <span>
         movies found
      </span>
    </div>
    <div
      class="moviesBlock-0-2-3"
    >
      <div
        class="movieItem-0-2-13"
      >
        <div
          class="control-0-2-14"
        />
        <a
          class="movieLink-0-2-17"
          href="/film/1000"
        >
          <img
            alt="testTitle"
            class="movieImg-0-2-18"
            src="testPoster"
          />
        </a>
        <div
          class="titleBlock-0-2-19"
        >
          <span
            class="title-0-2-20"
          >
            testTitle
          </span>
          <span
            class="date-0-2-21"
          >
            testDate
          </span>
        </div>
        <div
          class="ganre-0-2-22"
        >
          <span>
             
            ganre
             
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
